src/Vanilla/Cli/CliUtil.php

```diff
@@ -30,25 +31,53 @@ public static function write($msg) {
      * @param string $msg Exception's message.
      * @throws Exception
      */
-    public static function error($msg) {
-        throw new Exception($msg.PHP_EOL);
+    public static function fail($msg) {
+        $logger = new LogFormatter();
+
+        $logger
+            ->setDateFormat('')
+            ->error($msg);
+
+        throw new Exception();
     }
 
     /**
      * Get the addon.json file in the current directory.
      */
-    public static function getAddonJsonForCWD() {
-        $addonJsonPath = getcwd().'/addon.json';
+    public static function getAddonJsonForDirectory($directory) {
+        $addonJsonPath = $directory.'/addon.json';
 
         if (file_exists($addonJsonPath)) {
             $addonJson = json_decode(file_get_contents($addonJsonPath), true);
 
             if (json_last_error() !== JSON_ERROR_NONE) {
                 CliUtil::write("\n\nThere were some issues parsing your addon.json file. Please ensure that it is valid JSON");
-                CliUtil::error("\nError Type: ".json_last_error_msg());
+                CliUtil::fail("\nError Type: ".json_last_error_msg());
             }
 
             return $addonJson;
         }
     }
 }
+
+
+/**
+ * Like array_key_exists but for nested values.
+ *
+ * @param string $key A key in dot notation.
+ * @param mixed $collection The collection to check.
+ *
+ * @return bool
+ */
+function array_deep_key_exists($key, $collection) {
```

Comment:
I would put that in a separate file.

I'm also not seeing how this checks deeper levels of the collection.

I suggest you create a function file like we have in vanilla/vanilla and that you use valr with a `if !defined()`